go/private/sdk.bzl

```diff
@@ -121,6 +123,8 @@ _go_download_sdk = repository_rule(
         "sdks": attr.string_list_dict(),
         "urls": attr.string_list(default = ["https://dl.google.com/go/{}"]),
         "version": attr.string(),
+        "patch": attr.label(default = None),
+        "patch_strip": attr.int(default = 0),
```

Comment:
Let's follow the convention of `go_repository` rule:

* Having `patches` to be a label_list, so it can take multiple patches
* Having `patch_args` to be string_list, and defaults to `["-p0"]`